src/pages/workspaces/workspace/Data.js

```diff
@@ -1064,89 +1067,94 @@ const WorkspaceData = _.flow(
-          tableName: selectedData.tableName,
-          loadMetadata: () => loadSnapshotEntities(selectedData.snapshotName),
-          onUpdate: async newSnapshotName => {
-            await loadSnapshotMetadata()
-            setSelectedData({ type: workspaceDataTypes.snapshot, snapshotName: newSnapshotName })
-            forceRefresh()
-          },
-          onDelete: async () => {
-            await loadSnapshotMetadata()
-            setSelectedData(undefined)
-            forceRefresh()
-          }
-        })],
-        [workspaceDataTypes.entities, () => h(EntitiesContent, {
-          key: refreshKey,
-          workspace,
-          entityMetadata,
-          setEntityMetadata,
-          entityKey: selectedData.entityType,
-          activeCrossTableTextFilter,
-          loadMetadata,
-          deleteColumnUpdateMetadata,
-          forceRefresh
-        })],
-        [workspaceDataTypes.entitiesVersion, () => h(DataTableVersion, {
-          workspace,
-          version: selectedData.version,
-          onDelete: reportErrorAndRethrow('Error deleting version', async () => {
-            await deleteDataTableVersion(selectedData.version)
-            setSelectedData(undefined)
-          }),
-          onImport: reportErrorAndRethrow('Error importing version', async () => {
-            const { tableName } = await importDataTableVersion(selectedData.version)
-            await loadMetadata()
-            setSelectedData({ type: workspaceDataTypes.entities, entityType: tableName })
-          })
-        })],
-        [workspaceDataTypes.wds, () => wdsDataTableProvider && wdsReady && !_.isEmpty(wdsTypes.state) && h(WDSContent, {
-          key: refreshKey,
-          workspaceUUID: workspaceId,
-          workspace,
-          dataProvider: wdsDataTableProvider,
-          recordType: selectedData.entityType,
-          wdsSchema: wdsTypes.state
-        })]
-        )], () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } }, ['Currently, only a workspace creator can use data tables. If you were invited to this workspace, you can use the clone feature to create your own workspace to use data tables.']))
+        Utils.cond(
+          [createdBy === getUser()?.email || isGoogleWorkspace, () => Utils.switchCase(selectedData?.type,
+            [undefined, () => Utils.cond(
+              [isAzureWorkspace && !wdsReady, () => div({ style: { textAlign: 'center', lineHeight: '1.4rem', marginTop: '1rem', marginLeft: '5rem', marginRight: '5rem' } },
+                [icon('loadingSpinner'),
+                  ' Preparing your data tables, this may take a few minutes. ',
+                  Utils.cond(
+                    [!uploadingWDSFile, () => div({}, ['You can ', h(Link, { style: { marginTop: '0.5rem' }, onClick: () => setTroubleshootingWds(true) }, ['check the status']), ' of your data table service.'])],
+                    () => () => '')])],
```

Comment:
the only non-whitespace change from lines 1067-1078 is a change to the final catch-all case of the `Utils.cond`.
I changed `''` to `() => () => ''`.